amul_topicals.py

```diff
@@ -2,15 +2,30 @@
 import json
 import requests
 from bs4 import BeautifulSoup
+import urllib
+import os
+import sys
 
 HOST = 'http://www.amul.com'
 BASE_URL = HOST + '/m/amul-hits'
 
-def get_year_topicals(year_page):
+#current working directory
+CWD=os.getcwd()
+
+def get_year_topicals(year_page,location):
     "Gets image URLs of all topicals in the given page"
     topicals = {}
     i = 0
     j = 0
+
+    #ask user if they wish to download images by year
+    soup = BeautifulSoup(year_page, 'html.parser')
+    htable = soup.find('table')
+    heading = htable.find('strong').text
+    choice = get_UserYesOrNo("Do you wish to download images of - "+heading+"?", ['y','n'])
```

Comment:
I think here the user should also get a choice to exit the program completely. When I tried running the program, I notice that when I say no, the program keeps asking me for different years.

Reply:
At what point should I consider adding this feature? What I thought is that the user should get to go through all the years to figure out which ones to download and which ones to not. For example, say the user keeps saying no initially because the user wants to download images of a year whose query hasn't popped up yet, how do I handle this? Do I ask the user at every point they say no if they wish to continue further or exit?

How's this also for an idea? -  Maybe I could ask the user to input the year of which they want the images and get those done? We haven't really set specifications regarding this, so it could be done many ways.

Reply:
Yes, the latter, exactly. You can tell the user that topicals are available for a certain year range and then the user can tell the program which exact year (or years) they are interested in.

Reply:
I'll be working on making this change to get the years from the user, and only let the program run to fetch those.